pkg/validation/options.go

```diff
@@ -233,7 +233,10 @@ func parseProviderInfo(o *options.Options, msgs []string) []string {
 	p.ValidateURL, msgs = parseURL(o.ValidateURL, "validate", msgs)
 	p.ProtectedResource, msgs = parseURL(o.ProtectedResource, "resource", msgs)
 
-	// Make the OIDC Verifier accessible to all providers that can support it
+	// Make the OIDC options available to all providers that support it
+	p.AllowUnverifiedEmail = o.InsecureOIDCAllowUnverifiedEmail
+	p.EmailClaim = o.UserIDClaim
+	p.GroupsClaim = o.OIDCGroupsClaim
```

Comment:
Would be really nice to have some sort of provider constructor like we do in the session storage so we don't have to do this here 🤔

Reply:
Good idea. I can whip something up to help with this.

One of my next provider refactors I want to take on is getting a `Configure` method or something on each provider so the options they consume is much more clear in general.

Reply:
How do you feel about the placement of the globally available OIDC methods? `ProviderData` felt most global, but I'm not in love with it since the OIDC methods are a bit different than `ProviderData` type fields.

I also noticed both `internal_util` and `util` in that package, but I'll leave that consolidation for another PR.